tests/integration/test_movielens.py

```diff
@@ -20,113 +22,147 @@
 
     # Test if can handle different size of header columns
-    header = ["a"]
-    df = movielens.load_spark_df(spark, header=header)
-    assert len(df.columns) == len(header)
-
-    header = ["a", "b", "c", "d", "e"]
+    header = ["1", "2"]
+    schema = StructType([StructField("u", IntegerType())])
     with pytest.warns(Warning):
-        df = movielens.load_spark_df(spark, header=header)
-        assert len(df.columns) == 4
+        # Test if use schema when both schema and header are provided
+        df = movielens.load_spark_df(spark, size=size, header=header, schema=schema, local_cache_path=local_cache_path)
+        assert df.count() == num_samples
+        assert len(df.columns) == len(schema)
 
     # Test title load
-    df = movielens.load_spark_df(spark, size=size, title_col="Title")
-    assert len(df.columns) == 5
-    # Movie 1 is Toy Story
-    title = df.filter(col(DEFAULT_ITEM_COL) == 1).select("Title").limit(2).collect()
-    assert title[0][0] == title[1][0]
-    assert title[0][0] == title_example
-
-    # Test genres load
-    df = movielens.load_spark_df(spark, size=size, genres_col="Genres")
-    assert len(df.columns) == 5
-    # Movie 1 is Toy Story
-    genres = df.filter(col(DEFAULT_ITEM_COL) == 1).select("Genres").limit(2).collect()
+    df = movielens.load_spark_df(spark, size=size, local_cache_path=local_cache_path,
+                                 title_col="Title", genres_col="Genres", year_col="Year")
+    assert df.count() == num_samples
+    assert len(df.columns) == 7
+
+    # Get first two records of MovieID == 1, which is Toy Story
+    head = df.filter(col(DEFAULT_ITEM_COL) == 1).limit(2)
+
+    title = head.select("Title").collect()
+    assert title[0][0] == title[1][0]  # Check if two records are the same movie
+    assert title[0][0] == title_example  # Check if the record is Toy Story
+
+    genres = head.select("Genres").collect()
     assert genres[0][0] == genres[1][0]
     assert genres[0][0] == genres_example
 
-    # Test movie data load (not rating data)
-    df = movielens.load_spark_df(spark, size=size, header=None, title_col="Title", genres_col="Genres")
-    assert df.count() == num_movies
-    assert len(df.columns) == 3
-
-    # Test if can handle wrong size argument
-    with pytest.raises(ValueError):
-        movielens.load_spark_df(spark, size='10k')
-    # Test if can handle wrong cache path argument
-    with pytest.raises(ValueError):
-        movielens.load_spark_df(spark, local_cache_path='.')
+    year = head.select("Year").collect()
+    assert year[0][0] == year[1][0]
+    assert year[0][0] == year_example
 
-    # Test if use schema when both schema and header are provided
-    header = ["1", "2"]
-    schema = StructType([StructField("u", IntegerType())])
-    with pytest.warns(Warning):
-        df = movielens.load_spark_df(spark, header=header, schema=schema)
-        assert len(df.columns) == len(schema)
+    shutil.rmtree(local_cache_dir, ignore_errors=True)
```

Comment:
this is a little scary, and one reason working with TemporaryDirectories is safer, if someone accidentally changes local_cache_dir we could cause some damage here.

Reply:
Will change to use tempfile as you suggested!